topology/probes/k8s/graph.go

```diff
@@ -35,14 +37,36 @@ const (
 	detailsField = "K8s"
 )
 
-func newMetadata(ty, namespace, name string, details interface{}) graph.Metadata {
-	return graph.Metadata{
-		"Manager":    Manager,
-		"Type":       ty,
-		"Namespace":  namespace,
-		"Name":       name,
-		detailsField: common.NormalizeValue(details),
+// JoinName concatinates a namespace/name pair into a full-name
+func JoinName(namespace, name string) string {
+	return namespace + "/" + name
+}
+
+// splitName splits a full-name into a namespace/name pair
+func splitName(name string) (string, string, error) {
+	a := strings.Split(name, "/")
+	if len(a) == 2 {
+		return a[0], a[1], nil
+	}
+	return "", name, nil
+}
+
+// NewMetadata creates a k8s node base metadata struct
+func NewMetadata(manager, ty, name string, details interface{}) graph.Metadata {
```

Comment:
Please change the function to :
```
NewMetadata(manager, ty, details interface{}, name string, namespace ...string)
```

So that we can avoid joining/splitting string every time (and not having to export JoinName and SplitName at the same time)

Reply:
ok